frame/staking/src/pallet/mod.rs

```diff
@@ -184,8 +184,34 @@ pub mod pallet {
 		///
 		/// The changes to nominators are reported to this. Moreover, each validator's self-vote is
 		/// also reported as one independent vote.
+		///
+		/// To keep the load off the chain as much as possible, changes made to the staked amount
+		/// via rewards and slashes are not reported and thus need to be manually fixed by the
+		/// staker. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
+		///
+		/// Invariant: what comes out of this list will always be a nominator.
 		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
 
+		/// Something that provides a best-effort sorted list of targets aka electable validators,
+		/// used for NPoS election.
+		///
+		/// The changes to the approval stake of each validator are reported to this. This means any
+		/// change to:
+		/// 1. The stake of any validator or nominator.
```

Comment:
These rules are not really applied yet, right?

Reply:
this docs needs updating: you should explain the current status accurately, and also what's to come next.

To recap, as it stands now, this pallet makes not attempt at maintaining any updates in its `type TargetList`. Therefore, ONLY using `UseValidatorsMap` is valid. 

In a follow-up PR, you can add the update logic.

Reply:
ping.

Reply:
Github failed to update it for some reason, I've pushed a noop comment a while ago.